### ISSUE TITLE
ConcurrentModificationException in 1.8.3.4

### ISSUE THREAD

**Issue by Relaxing9** [snowgears/shopbugs#297]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Type of bug

Error in console

### Environment

```markdown
- Server Version: git-Paper-167 (MC: 1.18.1)
- Shop Version: 1.8.3.4
```

### Server startup log

https://paste.gg/p/anonymous/09c89b37d9da48b4ae0bb8f4a9bce5c2

### Shop config files

https://paste.gg/p/anonymous/73a4f5cba3004fffba1bff762ac8f923

### Error log (if applicable)

https://paste.gg/p/anonymous/8937c87d5b364c738bb010659a2e6190

### Bug description

Getting a CME on seemingly startup

### Steps to reproduce

Unknown

### Expected behaviour

No CME Displayed

### Actual behaviour

CME Displayed

**Comment by snowgears** [comment]
20.01 23:41:40] [Server] [WARN] [Shop] Plugin Shop v1.8.3.4 generated an exception while executing task 37
--
  | [20.01 23:41:40] [Server] java.util.ConcurrentModificationExceptionnull
  | [20.01 23:41:40] [Server] 	at java.util.HashMap$HashIterator.remove(HashMap.java:1611) ~[?:?]
  | [20.01 23:41:40] [Server] 	at com.snowgears.shop.handler.ShopHandler.processShopDisplaysNearPlayer(ShopHandler.java:423) ~[Shop-1.8.3.4.jar:?]
  | [20.01 23:41:40] [Server] 	at com.snowgears.shop.listener.DisplayListener$1.run(DisplayListener.java:64) ~[Shop-1.8.3.4.jar:?]
  | [20.01 23:41:40] [Server] 	at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.18.1.jar:git-Paper-167]
  | [20.01 23:41:40] [Server] 	at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.18.1.jar:git-Paper-167]
  | [20.01 23:41:40] [Server] 	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.18.1.jar:git-Paper-167]
  | [20.01 23:41:40] [Server] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
  | [20.01 23:41:40] [Server] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
  | [20.01 23:41:40] [Server] 	at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by snowgears** [comment]
99% sure I have fixed this in 1.8.3.6 (in progress). Please reopen if this happens again.
